### DISCUSSION TITLE
[request] SSO/SAML

### DISCUSSION THREAD

**Discussion by sonicaghi** [Kong/kong#7358] [Ideas and feature requests]

**Comment by lucamaraschi** [comment]
:+1:  :+1:

**Comment by mistertakai** [comment]
That would be great.

**Comment by mingfang** [comment]
+1

I have a very simplistic implementation working with OpenResty https://github.com/mingfang/docker-knginx/blob/master/saml/saml.lua

**Comment by dgomes** [comment]
+1

**Comment by t1tcrucible** [comment]
+1

**Comment by ddelazerda** [comment]
+1

**Comment by skymaster420** [comment]
+1

**Comment by subnetmarco** [comment]
@mingfang would you like to open a PR with your contribution? We could test it and include it as an official plugin in Kong.

**Comment by ddelazerda** [comment]
@thefosk it looks like that implementation requires nodejs.

**Comment by peacemakr** [comment]
+1 

still trying to figure out why OAuth2 plugin can't help you with authentication? could anyone be so kind to shed some light on it (in the context of Kong)? I know the basics like Oauth2 is for AuthN (authorization) whereas SAML/OpenID etc are for authX (authentication).

However, with scopes like Login, profile etc you can authenticate a user via a Kong server and get the user data, no?

**Comment by shashanksinghin** [comment]
+1

**Comment by chenrui333** [comment]
👍 👍 👍
